File: monitor-manage/internal/service/send/send.go
package send

import (
	"gitee.com/zekeGitee_admin/tx_gdut_monitor/monitor-manage/internal/model"
)

type ISend interface {
	// GetAllConfigs 获取所有配置
	GetAllConfigs() ([]model.SendConfig, error)
	// GetConfigs 获取配置列表
	GetConfigs(ip string, local string) ([]model.SendConfig, error)
	// AddConfig 新增配置
	AddConfig(Ip string, Local string, SendType int32, Level int32, Config string) error
	// Update 更新发送服务配置
	Update(id int32, IP string, Local string, SendType int32, Level int32, Config string) error
	// Del 删除指定id的配置
	Del(ip string, local string, id int32) error
	// Init 初始化
	Init()
}

File: monitor-web/monitor-txui/src/api/send.js
import { fetchGet, fetchPost } from '@/request.js'

// 获取指定agent的判定规则信息
export function GetSendConfigs (ip, local) {
  let url = '/send/info/' + ip + '/' + local
  return fetchGet(url, {})
}

export function AddSendConfig (ip, local, sendType, level, config) {
  console.log(ip, local, sendType, level, config)
  let url = '/send/info/' + ip + '/' + local
  return fetchPost(url, {
    'sendType': sendType,
    'level': level,
    'config': config
  })
}

// 更新agent的指定id判定规则信息
export function UpdateSendConfig (ip, local, id, sendType, level, config) {
  console.log(ip, local, id, sendType, level, config)
  let url = '/send/update/' + ip + '/' + local + '/' + id
  return fetchPost(url, {
    'sendType': sendType,
    'level': level,
    'config': config
  })
}

// 删除agent的指定id判定规则信息
export function DelSendConfig (ip, local, id) {
  let url = '/send/del/' + ip + '/' + local + '/' + id
  return fetchPost(url, {})
}
